Add new docs section for DNS Analytics (#557)
dnsimple/dnsimple-developer: content/v2/dns-analytics.markdown

---
title: DNS Analytics API | DNSimple API v2
excerpt: This page documents the DNS Analytics API v2 endpoint.
---

# DNS Analytics

* TOC
{:toc}

<note>
  This API is currently in Public Beta. During the Public Beta period changes may occur at any time.
</note>

## Query DNS Analytics data {#query}

Queries and returns DNS Analytics data available for the provided query parameters 

~~~
GET /:account/dns_analytics
~~~

### Parameters

Name | Type | Description
-----|------|------------
`:account` | `integer` | The account ID

### Groupings

Name | Description
-----|------------
`:zone_name` | Group rows by zone name
`:date` | Group rows by date

No groupings will be applied by default unless specified.
Multiple groupings can be provided, separated by a comma.

<info>
Applying groupings to the query will add columns to the response payload.
</info>

### Filters

Name | Default value                   | Description
-----|---------------------------------|------------

> `:start_date` | 31 days before the current date | Only include results up to the provided date

Reviewer: ```suggestion
`:start_date` | 31 days before the current date | Only include results starting from the provided date
```

> `:end_date` | 31 days before the current date | Only include results up to the provided date

Author: Indeed. I'll fix this!

> `:end_date` | 31 days before the current date | Only include results up to the provided date

Author: See https://github.com/dnsimple/dnsimple-developer/pull/557/commits/ac965dfa66cc302dfefec38f2a3916a89a158552

> `:end_date` | The day before the current date | Only include results starting from the provided date

Reviewer: ```suggestion
`:end_date` | The day before the current date | Only include results up to the provided date
```

<info>

> Both filters **must be used together**, and date values must be in **ISO 8601 format** (e.g. `2024-08-01`). Both dates are **inclusive**.

Author: See https://github.com/dnsimple/dnsimple-developer/pull/557/commits/ac965dfa66cc302dfefec38f2a3916a89a158552

I finally opted to remove the whole sentence while we discuss this.

</info>

<warning>
The maximum allowed date range is 31 days.
</warning>

### Sorting

For general information about sorting, please refer to the [main guide](/v2/#sorting).

Name | Description
-----|------------
`zone_name` | Sort rows by zone name
`date` | Sort rows by date
`volume` | Sort rows by query volume

The default sorting policy is by ascending `date`, then ascending `zone_name`.

### Pagination

Pagination is supported in this endpoint:

- The default page size (`per_page` query parameter) is 1,000
- The maximum supported page size (`per_page` query parameter) is 10,000

> - Please refer to the [main guide](/v2/#pagination)

Author: Good idea! I'll do it

> - Please refer to the [main guide](/v2/#pagination)

Author: See https://github.com/dnsimple/dnsimple-developer/pull/557/commits/ac965dfa66cc302dfefec38f2a3916a89a158552
